src/Statistics.h

```diff
@@ -133,29 +25,43 @@ class cStatManager
 
 	// tolua_begin
 
-	/** Return the value of the specified stat. */
-	StatValue GetValue(const eStatistic a_Stat) const;
+	/** Return the value of the specified stat as enum. */
+	StatValue GetValue(const eStatistic &a_Stat) const;
+
+	/** Return the value of the specified stat as string. */
+	StatValue GetValue(const AString &a_Stat) const;
 
-	/** Set the value of the specified stat. */
-	void SetValue(const eStatistic a_Stat, const StatValue a_Value);
+	/** Set the value of the specified stat as enum. */
+	void SetValue(const eStatistic &a_Stat, const StatValue a_Value);
+
+	/** Set the value of the specified stat as string. */
+	void SetValue(const AString &a_Stat, const StatValue a_Value);
 
 	/** Reset everything. */
 	void Reset();
 
-	/** Increments the specified stat.
-	Returns the new value.
-	*/
-	StatValue AddValue(const eStatistic a_Stat, const StatValue a_Delta = 1);
+	/** Increments the specified stat as enum. Returns the new value. */
+	StatValue AddValue(const eStatistic &a_Stat, const StatValue a_Delta = 1);
 
-	// tolua_end
 
-private:
+	/** Increments the specified stat as string. Returns the new value. */
+	StatValue AddValue(const AString &a_Stat, const StatValue a_Delta = 1);
+
+	// tolua_end
+	/** Type -> Name */
+	static const AString & GetName(const eStatistic a_Type);
 
-	StatValue m_MainStats[statCount];
+	/** returns the prerequisite for a achievement as String */
+	static const AString & GetPrerequisite(const AString & a_Type);
 
-	// TODO 10-05-2014 xdot: Use, mine, craft statistics
+	/** returns the count of dirty statistics for the given map aka version */
+	int GetDirtyStatistics(const std::map<AString, int> & map) const;
 
+	/** returns the count of dirty statistics for the given map aka version */
+	int GetDirtyStatistics(const AStringMap & map) const;
 
+private:
+	std::map<AString, int> m_MainStats= {{}};
```

Comment:
We should not use strings or stringmaps in the internal state. The list of statistics are static data and shouldn't impose a runtime allocation penalty.

Reply:
The array used is now directly created for each Object and only is map<enum class, int>

Reply:
see later on - now using somethin else